crates/rolldown/src/bundler/bundler.rs

```diff
@@ -88,10 +69,13 @@ impl Bundler {
     }
 
     self.closed = true;
-    self.bundle_factory.plugin_driver.close_bundle().await?;
+    // TODO: This infers that `close_bundle` will not be called if there is no bundle happened. Is this expected?
```

Comment:
This's a UB in Rollup. There's no way to only create a `bundle` in rollup without triggering the `buildStart` hook. For API like `rollup(..)/watch(..)`, build stage hooks will surely be executed.

So for rollup, the quesion that "should we trigger `closeBundle` hook if there's no build/bundle happens" doesn't exist in the rollup world.

Notice: `Bundler`, unlike `ClassicBundler`, is designed for `watch/dev` mode.

@sapphi-red cc